BreadPlayer.Views.UWP/ViewModels/ShellViewModel.cs

```diff
@@ -350,40 +353,41 @@ private void SetNowPlayingSong()
         {
-            var playingCollection = GetPlayingCollection();
-            NowPlayingQueue = playingCollection;
+            var playingCollection = Playlist;
+            NowPlayingQueue = Playlist;
+            Mediafile toPlayFile = null;
             if (playingCollection != null && playingCollection.Any())
             {
                 try
                 {
-                    int indexOfCurrentlyPlayingFile = -1;
-                    if (playingCollection.Any(t => t.State == PlayerState.Playing))
+                    if (IsSourceGrouped)
                     {
-                        indexOfCurrentlyPlayingFile = playingCollection.IndexOf(playingCollection.FirstOrDefault(t => t.State == PlayerState.Playing));
+                        toPlayFile = GetNextSongInGroup();
                     }
-
-                    Mediafile toPlayFile = null;
-                    if (Shuffle)
+                    else
                     {
-                        if (_shuffledList.Count < playingCollection.Count 
-                            || _shuffledList == null
-                            || indexOfCurrentlyPlayingFile > playingCollection.Count - 2)
+                        int indexOfCurrentlyPlayingFile;
+                        if (playingCollection.Any(t => t.State == PlayerState.Playing))
+                        {
+                            indexOfCurrentlyPlayingFile =
+                                playingCollection.IndexOf(
+                                    playingCollection.FirstOrDefault(t => t.State == PlayerState.Playing));
+                        }
+                        else
                         {
-                            _shuffledList = await ShuffledCollection();
-                            
                             indexOfCurrentlyPlayingFile = 0;
                         }
-                        toPlayFile = _shuffledList?.ElementAt(indexOfCurrentlyPlayingFile + 1);
-                    }
-                    else if (IsSourceGrouped)
-                    {
-                        toPlayFile = GetNextSongInGroup();
-                    }
-                    else
-                    {
-                        toPlayFile = indexOfCurrentlyPlayingFile <= playingCollection.Count - 2 && indexOfCurrentlyPlayingFile != -1 ? playingCollection.ElementAt(indexOfCurrentlyPlayingFile + 1) : Repeat == "Repeat List" || isNext ? playingCollection.ElementAt(0) : null;
+                        var indexToPlay = indexOfCurrentlyPlayingFile;
+                        do
+                        {
+                            indexToPlay += 1;
+                            if (indexToPlay > playingCollection.Count - 1)
+                            {
+                                indexToPlay = 0;
+                            }
+                            toPlayFile = playingCollection[indexToPlay];
+                        } while (Shuffle && toPlayFile.SkipOnShuffle);
```

Comment:
Won't a `do-while` loop just keep running until the `Shuffle` is enabled? A `do while` loop is basically in pseudo-language: `keep doing this while this is true` so you are basically raising the `indexToPlay` in an infinite loop up until the user turns either the `SkipOnShuffle` or `Shuffle` property off.

You can simply use a `if statement` for this.